File: src/main/java/dev/psygamer/wireframe/block/state/property/BooleanBlockProperty.java
package dev.psygamer.wireframe.block.state.property;

import com.google.common.collect.ImmutableSet;

import java.util.Optional;

public class BooleanBlockProperty extends BlockProperty<Boolean> {
	
	public BooleanBlockProperty(final String propertyName) {
		super(propertyName, false);
	}
	
	@Override
	public Optional<Boolean> getValue(final String valueName) {
		if (valueName.equals("true"))
			return Optional.of(true);
		if (valueName.equals("false"))
			return Optional.of(false);
		
		return Optional.empty();
	}
	
	@Override
	public String getValueName(final Boolean value) {
		return value.toString();
	}
	
	@Override
	public ImmutableSet<Boolean> getPossibleValues() {
		return ImmutableSet.of(true, false);
	}
}

File: src/main/kotlin/dev/psygamer/wireframe/api/block/BooleanBlockProperty.kt
package dev.psygamer.wireframe.api.block

import java.util.*

class BooleanBlockProperty(propertyName: String) : BlockProperty<Boolean>(propertyName, false) {

	override fun getValue(valueName: String): Optional<Boolean> {
		return when (valueName) {
			"true" -> Optional.of(true)
			"false" -> Optional.of(false)

			else -> Optional.empty()
		}
	}

	override fun getValueName(value: Boolean): String {
		return value.toString()
	}

	override val possibleValues: Set<Boolean>
		get() = setOf(true, false)
}